monix-eval/shared/src/main/scala/monix/eval/Coeval.scala

```diff
@@ -394,6 +394,20 @@ object Coeval {
   def fromTry[A](a: Try[A]): Coeval[A] =
     Attempt.fromTry(a)
 
+  /** Aquires a `Closeable` and closes it after the `Coeval` has been run.
+    */
+  def bracket[R <: java.io.Closeable](aquire: => R): Coeval[R] = defer {
+    val resource: R = aquire
+    now(resource).doOnFinish(_ => delay(resource.close))
```

Comment:
Then perhaps the following would be more generally useful:

````
    def bracket[R, A](resource: => R)(cleanup: R => ())(f: R => A): Coeval[A] =
  Coeval.defer {
    val ref = resource
    val close = Task.eval(cleanup(ref)))
    Coeval.eval(f(ref)).doOnFinish(_ => close)
  }
````

another flavor with a `cleanup: R => Task[Unit]`  would be useful too.

Lastly, in a signature that takes something from java.lang, `AutoClosable` should be preferred over `Closeable` as it is a supertype.

````
def using[R <: java.lang.AutoClosable, A](resource: => R)(f: R => A): Coeval[A] = bracket(resource)(r => r.close())(f)
````

Another question would be what to name variants that take a `R => Coeval[A]` rather than a `R => A` ?    Both are probably equally useful.